tests/node_test.cc

```diff
@@ -683,6 +683,93 @@ TEST_CASE("Node is checked for 2D case", "[node][2D]") {
       // TODO Assert: REQUIRE_NOTHROW(node->update_external_force(false, 1,
       // force_bad));
 
+      SECTION("Check absorbing constraint") {
+        // Set external force to 0
```

Comment:
Could you use an already existing function, so you don't have to redefine and check all the momenta and forces?